README.md

```diff
@@ -1,3 +1,448 @@
 # GraphQL.DI
 
 [![NuGet](https://img.shields.io/nuget/v/Shane32.GraphQL.DI.svg)](https://www.nuget.org/packages/Shane32.GraphQL.DI) [![Coverage Status](https://coveralls.io/repos/github/Shane32/GraphQL.DI/badge.svg?branch=master)](https://coveralls.io/github/Shane32/GraphQL.DI?branch=master)
+
+## Overview
+
+GraphQL.DI enhances GraphQL.NET's code-first approach by providing dependency injection support for field resolvers through the `DIObjectGraphBase` class. This enables a more maintainable and testable approach to building GraphQL APIs by allowing services to be injected directly into your field resolver classes.
+
+## Type-First vs Code-First in GraphQL.NET
+
+### Type-First Approach
+
+The type-first approach in GraphQL.NET automatically infers the GraphQL schema from your C# models:
+
+```csharp
+// The GraphQL type is automatically inferred from the model
+public class Todo
+{
+    public int Id { get; set; }
+    public string Title { get; set; }
+    public Person CompletedBy { get; set; }
+}
+```
+
+### Code-First Approach
+
+The traditional code-first approach in GraphQL.NET uses `ObjectGraphType<T>`:
+
+```csharp
+public class TodoType : ObjectGraphType<Todo>
+{
+    public TodoType(IRepository repository)
+    {
+        Field(x => x.Id);
+        Field(x => x.Title);
+        Field<PersonType>("completedBy")
+            .Resolve(context => repository.GetPersonById(context.Source.CompletedByPersonId));
+    }
+}
+```
+
+However in the above sample, the `IRepository` service must be registered as a singleton, which can
+lead to issues with scoped services. GraphQL.DI solves this problem by allowing scoped services to
+be injected directly into the field resolver classes, for better coding patterns and readability.
+
```

Comment:
I feel like i don't really get much understanding between why i should use one or the other.  It just kind of states that there could be issues using the code-first approach, but then your library literally just "handles it"  I think we should figure out why we should use one or the other, then consider switching all of our graphql libraries to what we think is better.